pkg/federate/kubefed/cluster.go

```diff
@@ -246,62 +261,50 @@ func newKubeFedClusterListerWatcher(kubeFedConfig *rest.Config) (cache.ListerWat
 }
 
 func (c *clusterWatcher) onAdd(clusterID string, kubeConfig *rest.Config) {
```

Comment:
Since you already have a struct for the event it makes it much simpler and you can just have one method enqueue which would receive the correct "event struct" and enqueue it.

E.G.
func (c *clusterWatcher) enqueue(event interface{}) {
    c.eventQueue.Add(event)
}

Then for calling when cluster added:
clusterWatcher.enqueue(&clusterAdded{clusterID, kubeConfig})

Reply:
yeah - either way. It's a bit of a remnant of the prior implementation but I think it reads a bit better at the call sites.

Reply:
Actually thinking about this more, it would be more efficient for the caller to create a single `clusterAdded` instance and pass to each `clusterWatcher` as you suggest instead of each `clusterWatcher` creating it's own instance. Also I'm going to rename `runnable` to 'clusterEvent` for clarity.